hyperspy/datasets/artificial_data.py

```diff
@@ -356,3 +363,231 @@ def get_atomic_resolution_tem_signal2d():
 
     s = Signal2D(image)
     return s
+
+
+def get_luminescence_spectrum_nonuniform(uniform=False, add_baseline=False, add_noise=True,random_state=None):
```

Comment:
To reduce duplication of code, an alternative would be to adapt and re-use `get_luminescence_map_nonuniform` use to covert this function. Maybe make rename `get_luminescence_map_nonuniform` to `get_luminescence_signal` and add a `navigation_dimension` argument to `get_luminescence_signal` to cover the behaviour of both functions?

Reply:
I agree. Will do